addons/website_forum/tests/test_performance.py

```diff
@@ -0,0 +1,71 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+
+from odoo.addons.website.tests.test_performance import UtilPerf
+
+
+class TestForumPerformance(UtilPerf):
+    def setUp(self):
+        super().setUp()
+        self.user_employee = self.env['res.users'].with_context({'no_reset_password': True}).create({
+            'name': 'Armande Employee',
+            'login': 'Armande',
+            'email': 'armande.employee@example.com',
+            'karma': 0,
+            'groups_id': [(6, 0, [self.env.ref('base.group_user').id])]
+        })
+        self.forum = self.env['forum.forum'].create({'name': 'TestForum'})
+        self.post = self.env['forum.post'].create({
+            'name': 'TestQuestion',
+            'content': "Who's afraid of Virginia Wolf?",
+            'forum_id': self.forum.id,
+            'tag_ids': [(0, 0, {'name': 'Tag2', 'forum_id': self.forum.id})]
+        })
+        self.forum_website_url = self.forum._compute_website_url()
+
+    def test_perf_sql_forum_standard_data(self):
+        number_of_queries = self._get_url_hot_query(self.forum_website_url)
+        self.assertEqual(number_of_queries, 24)
+        number_of_queries = self._get_url_hot_query(self.forum_website_url, cache=False)
+        self.assertLessEqual(number_of_queries, 28)
+        number_of_queries = self._get_url_hot_query(self.post.website_url)
+        self.assertEqual(number_of_queries, 20)
+        number_of_queries = self._get_url_hot_query(self.post.website_url, cache=False)
+        self.assertLessEqual(number_of_queries, 24)
+
+    def test_perf_sql_forum_scaling_answers(self):
+        self.env['forum.tag'].create([{
+            'name': 'Forum Post Test Tag %s' % i, 'forum_id': self.forum.id
+        } for i in range(20)])
+        answers = self.env['forum.post'].create([{
+            'name': f"TestAnswer {i}",
+            'content': "You" if i == 4 else f"Me {i}",
+            'is_correct': i == 4,
+            'forum_id': self.forum.id,
+            'parent_id': self.post.id,
+        } for i in range(50)])
+        self.env['forum.post.vote'].create([
+            {
+                'vote': str(random.randint(-1, 1)),
+                'post_id': answers[i].id,
+                'user_id': self.user_employee.id,
+            }
+            for i in range(20)
+        ])
+        number_of_queries = self._get_url_hot_query(self.post.website_url)
+        self.assertEqual(number_of_queries, 23)
+        number_of_queries = self._get_url_hot_query(self.post.website_url, cache=False)
+        self.assertLessEqual(number_of_queries, 27)
+
+    def test_perf_sql_forum_scaling_posts(self):
+        self.env['forum.post'].create([{
```

Comment:
Not sure to know all impact of "_get_url_hot_query", but would probably flush before calling it